### PR TITLE
Trying to PR the tropea version

### PR DESCRIPTION
**Pull request by matteobecchi**
Requested Reviewers: @andrewtarzia

Switching from importing onion-clustering to importing the new version, tropea-clustering. 
The main difference is that now tropea-clustering offers two functions (onion_uni and onion_multi) which take as input the data array and returns as output
- a list of the Gaussian states
- the array with the label for each signal window

The original reasons for this change were to avoid onion printing figures and files all over the user's PC, and avoid requiring Matplotlib in the pytests. 

The user will have to switch to the new input/output format. There are examples showing how to do everything, together with the "plot" module.

### CODE REVIEW

File: src/dynsight/_internal/utilities/utilities.py [hunk 1]
```diff
@@ -1,11 +1,7 @@
-from typing import Literal
-
 import numpy as np
```
Comment (andrewtarzia) [inline src/dynsight/_internal/utilities/utilities.py:1]: Revert this file to LHS.

File: src/dynsight/analysis.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 """analysis package."""
```
Comment (andrewtarzia) [inline src/dynsight/analysis.py:2]: Revert to LHS.

File: src/dynsight/utilities.py [hunk 3]
```diff
@@ -1,11 +1,7 @@
 """utilities package."""
```
Comment (andrewtarzia) [inline src/dynsight/utilities.py:2]: Revet to LHS

File: tests/data_processing/test_spatialaverage.py [hunk 4]
```diff
@@ -8,10 +8,16 @@
 
 
 def test_spatialaverage() -> None:
-    original_dir = Path(__file__).resolve().parent
```
Comment (andrewtarzia) [inline tests/data_processing/test_spatialaverage.py:11]: Revert to LHS.

File: tests/data_processing/trajectory/.test_spavg.xtc_offsets.lock
Comment (andrewtarzia) [inline tests/data_processing/trajectory/.test_spavg.xtc_offsets.lock:1]: Delete this from your local.

File: tests/data_processing/trajectory/.test_spavg.xtc_offsets.npz
Comment (andrewtarzia) [inline tests/data_processing/trajectory/.test_spavg.xtc_offsets.npz:1]: Delete this from your local.

File: tests/data_processing/trajectory/test_spavg.gro
Comment (andrewtarzia) [inline tests/data_processing/trajectory/test_spavg.gro:1]: Delete this from your local.

File: tests/data_processing/trajectory/test_spavg.xtc
Comment (andrewtarzia) [inline tests/data_processing/trajectory/test_spavg.xtc:1]: Delete this from your local.

File: docs/source/analysis.rst [hunk 5]
```diff
@@ -6,4 +6,3 @@ analysis
   :maxdepth: 1
 
   compute_entropy_gain <_autosummary/dynsight.analysis.compute_entropy_gain>
```
Comment (andrewtarzia) [inline docs/source/analysis.rst:8]: Revert this file.

**Review by andrewtarzia — APPROVED** [review]

**Comment by matteobecchi** [comment]
meme_PR

**Pull request merged**